File: B/PHP/b010.php
<?php
$XA = array();
$XB = array();

$tmp = trim(fgets(STDIN));
$tmp = str_replace(array("\r\n","\r","\n"), '', $tmp);
$tmp = explode(" ", $tmp);
$T = $tmp[0];
$U = intval($tmp[1]);

$tmp = trim(fgets(STDIN));
$tmp = str_replace(array("\r\n","\r","\n"), '', $tmp);
$tmp = explode(" ", $tmp);
foreach ($tmp as $x) { array_push($XA,intval($x)); }

$tmp = trim(fgets(STDIN));
$tmp = str_replace(array("\r\n","\r","\n"), '', $tmp);
$tmp = explode(" ", $tmp);
foreach ($tmp as $x) { array_push($XB,intval($x)); }

if ($T == "A") {
    $pass = $XA[$U-1];
    sort($XB);
    $b = $XB[9];
    $flg = false;
    $num = 1;
    foreach ($XA as $a) {
        if ($a>=55 && $a>$b && $a>$pass) {
            echo $num . "\n";
            $flg = true;
        }
        $num += 1;
    }
    if ($flg == false) { echo "None\n"; }
} else {
    $pass = $XB[$U-1];
    sort($XA);
    $a = $XA[1];
    $flg = false;
    $num = 1;
    foreach ($XB as $b) {
        if ($b<=55 && $b<$a && $b<$pass) {
            echo $num . "\n";
            $flg = true;
        }
        $num += 1;
    }
    if ($flg == false) { echo "None\n"; }
}
?>

File: B/C++/b010.cpp
#include <algorithm>
#include <iostream>
using namespace std;
int main() {

    int     XA[11];
    int     XB[11];
    char    T[4];
    int     U,pass,flg,num,a,b;
    int     i;

    cin >> T;
    cin >> U;
    for (i=0; i<11; i++) { cin >> XA[i]; }
    for (i=0; i<11; i++) { cin >> XB[i]; }

    if (strcmp(T,"A") == 0) {
        pass = XA[U-1];
        sort(XB, XB+11);
        b = XB[9];
        flg = 0;
        num = 1;
        for (i=0; i<11; i++) {
            a = XA[i];
            if (a>=55 && a>b && a>pass) {
                cout << num << "\n";
                flg = 1;
            }
            num += 1;
        }
        if (flg == 0) { cout << "None\n"; }
    } else {
        pass = XB[U-1];
        sort(XA, XA+11);
        a = XA[1];
        flg = 0;
        num = 1;
        for (i=0; i<11; i++) {
            b = XB[i];
            if (b<=55 && b<a && b<pass) {
                cout << num << "\n";
                flg = 1;
            }
            num += 1;
        }
        if (flg == 0) { cout << "None\n"; }
    }
}
